core/workspace.js

```diff
@@ -162,19 +154,37 @@ Blockly.Workspace.prototype.MAX_UNDO = 1024;
 Blockly.Workspace.prototype.connectionDBList = null;
 
 /**
- * Adds cursor for keyboard navigation.
- * @return {!Blockly.Cursor} Cursor for keyboard navigation.
+ * Sets the cursor for keyboard navigation.
+ * @param {Blockly.Cursor} cursor The cursor used to navigate around the blockly
+ *     ast for keyboard navigation.
+ */
+Blockly.Workspace.prototype.setCursor = function(cursor) {
+  this.cursor = cursor;
+};
+
+/**
+ * Sets the marker for keyboard navigation.
+ * @param {Blockly.MarkerCursor} marker The marker used to mark a location for
+ *     keyboard navigation.
+ */
+Blockly.Workspace.prototype.setMarker = function(marker) {
+  this.marker = marker;
```

Comment:
Same as cursor, this.marker needs to be defined in the constructor, and potentially private.

Reply:
done